### PR TITLE
Fix/ledger improvements

### PR DESCRIPTION
**Pull request by antonijzelinskij**
#8695043zg

### CODE REVIEW

File: feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/presentation/account/connect/generic/selectLedger/SelectLedgerGenericImportViewModel.kt [hunk 1]
```diff
@@ -37,12 +35,13 @@ class SelectLedgerGenericImportViewModel(
...
         ledgerMessageCommands.value = LedgerMessageCommand.Hide.event()
 
-        val universalAccount = substrateApplication.getUniversalAccount(device, confirmAddress = false)
-        val payload = PreviewImportGenericLedgerPayload(
-            account = universalAccount.toGenericParcel(),
-            deviceId = device.id
+        val payload = SelectLedgerAddressPayload(
+            deviceId = device.id,
+            chainId = getPreviewBalanceChainId()
         )
```
Comment (antonijzelinskij) [inline feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/presentation/account/connect/generic/selectLedger/SelectLedgerGenericImportViewModel.kt:41]: I think we can keep payload the same but pass chainId of Polkadot to show amount in DOT

**Comment by stepanLav** [comment]
Task linked: [CU-8695043zg Android. Ledger Legacy improvements](https://app.clickup.com/t/8695043zg)

File: feature-ledger-api/src/main/java/io/novafoundation/nova/feature_ledger_api/sdk/application/substrate/LedgerSubstrateAccount.kt [hunk 2]
```diff
@@ -3,6 +3,7 @@ package io.novafoundation.nova.feature_ledger_api.sdk.application.substrate
 import io.novasama.substrate_sdk_android.encrypt.EncryptionType
 
 class LedgerSubstrateAccount(
+    val index: Int,
```
Comment (valentunn) [inline feature-ledger-api/src/main/java/io/novafoundation/nova/feature_ledger_api/sdk/application/substrate/LedgerSubstrateAccount.kt:6]: There is no need for that - index is encoded in derivation path

File: feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/presentation/account/common/selectAddress/SelectAddressLedgerViewModel.kt [hunk 3]
```diff
@@ -39,7 +39,7 @@ import kotlinx.coroutines.withContext
 
 abstract class SelectAddressLedgerViewModel(
     private val router: LedgerRouter,
-    private val interactor: SelectAddressLedgerInteractor,
+    internal val interactor: SelectAddressLedgerInteractor,
```
Comment (valentunn) [inline feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/presentation/account/common/selectAddress/SelectAddressLedgerViewModel.kt:42]: Maybe it should be `protected`?

File: feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/presentation/account/connect/generic/selectAddress/SelectAddressImportGenericLedgerViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    chainRegistry = chainRegistry,
+    messageFormatter = messageFormatter
+) {
+
+    override val needToVerifyAccount = false
+
+    override fun onAccountVerified(account: LedgerAccountWithBalance) {
+        launch {
+            val device = interactor.getDevice(payload.deviceId)
+            val universalAccount = substrateApplication.getUniversalAccount(device, account.index, confirmAddress = false)
```
Comment (valentunn) [inline feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/presentation/account/connect/generic/selectAddress/SelectAddressImportGenericLedgerViewModel.kt:41]: Why do we need to fetch account once again? All necessary data should be in `account`

File: feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/sdk/application/substrate/LedgerAppCommon.kt [hunk 5]
```diff
@@ -78,7 +78,7 @@ object SubstrateLedgerAppCommon {
         return junctions.serializeInLedgerFormat()
     }
 
-    fun parseAccountResponse(raw: ByteArray, requestDerivationPath: String): LedgerSubstrateAccount {
+    fun parseAccountResponse(accountIndex: Int, raw: ByteArray, requestDerivationPath: String): LedgerSubstrateAccount {
```
Comment (valentunn) [inline feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/sdk/application/substrate/LedgerAppCommon.kt:81]: This seems to be not needed

**Review by valentunn — APPROVED** [review]

**Pull request merged**